lib/analytics.js

```diff
@@ -14,16 +14,32 @@ if (process.env.REDISTOGO_URL) {
 }
 
 let analytics = {};
+let autosaveIntervalId;
 
 const analyticsAutoSaveFileName = process.env.SHIELDS_ANALYTICS_FILE || './analytics.json';
-const analyticsAutoSavePeriod = 10000;
-setInterval(function analyticsAutoSave() {
+
+function performAutosave() {
+  const contents = JSON.stringify(analytics);
   if (useRedis) {
-    redis.set(analyticsAutoSaveFileName, JSON.stringify(analytics));
+    redis.set(analyticsAutoSaveFileName, contents);
   } else {
-    fs.writeFileSync(analyticsAutoSaveFileName, JSON.stringify(analytics));
+    fs.writeFileSync(analyticsAutoSaveFileName, contents);
   }
-}, analyticsAutoSavePeriod);
+}
+
+function scheduleAutosaving() {
```

Comment:
We should be consistent about `autosave` (lowercase S) vs `autoSave` (capital S) in this file.